### ISSUE TITLE
SSMP2 Communication - Invalid Protocol header Ubuntu and Windows

### ISSUE THREAD

**Issue by Matthew-Cherry87** [Comer352L/FreeSSM#29]
Hi all,

Greatly appreciate any feedback or suggestions on resolving this issue i'm encountering

Compiled the latest FreeSSM master, but receive the error 'invalid protocol header' when attempting to connect via serial to com port. See attached screen-shot from the FreeSSM debug for further information and the debug output. This error occurs under both Ubuntu and Windows 10. 

This issue is specific to the current master. FreeSSM 1.25 has no issues successfully connecting via the same com port and communicating with the ECU from either operating system. Vehicle is a 2006 AUDM Impreza STi. 

I've tried compiling the master release with QT 5.8 using both mingw32 7.3 and 5.3 in both 64bit and 32bit flavours for both Win10 and Ubuntu. No difference between OS, QT, x86, x64, or mingw releases. cannot successfully obtain any connection to ECU. Followed recommended process of qmake, make release, make translation, make install. No issues compiling either release or debug. 
 
In short, can not connect with current master, can with older release. 

Regards,

Matthew
![FreeSSMDebugError](https://user-images.githubusercontent.com/49865102/56496346-326ba480-6538-11e9-8e2c-572a045d4642.png)

**Comment by Comer352L** [comment]
Which interface are you using ?
You have tested v1.25 so I assume you are using a serial pass-through interface, right ?
Is it a simple one or does it emulate the pass-through mode with a microprocessor ?

**Comment by Comer352L** [comment]
Please email me directly (see credits), I will send you a debugging patch.

**Comment by Matthew-Cherry87** [comment]
Hi Comer, 

Yes, using serial pass-through interface with a BTSSM Bluetooth adapter. I assume it has a microprocessor for pass-through, as the outbound com port is labeled 'SPP Dev'. 

I'll send you an email now, thanks for your help greatly appreciated.

**Comment by Matthew-Cherry87** [comment]
Email sent. :)

**Comment by Comer352L** [comment]
Fixed now in branch "master".
